docker/opensource.md

```diff
@@ -1,6 +1,106 @@
 # Open Source
 {id: open-source}
 
+## TOML Kit
+* [tomlkit](https://github.com/sdispater/tomlkit)
+
+Steps to run tests on a docker container:
+```
+git clone --recurse-submodules https://github.com/sdispater/tomlkit.git
+cd tomlkit
+docker run -it --name toml -w /opt -v$(pwd):/opt python:3.11 bash
+pip install poetry
+pip install pytest
+poetry install
+poetry run pytest -q tests
+```
+
+## Dialogy
+* [Dialogy](https://github.com/skit-ai/dialogy)
+
+Steps to run tests on a docker container:
+```
+git clone https://github.com/skit-ai/dialogy.git
+cd dialogy
+docker run -it --name dialogy -w /opt -v$(pwd):/opt python:3.11 bash
+pip install poetry
+poetry install
+make install
+```
+
+## Teiphy
+* [Teiphy](https://github.com/jjmccollum/teiphy)
+
+Steps to run tests on a docker container:
+```
+git clone https://github.com/jjmccollum/teiphy.git
+cd teiphy
+docker run -it --name teiphy -w /opt -v$(pwd):/opt python:3.11 bash
+pip install poetry
+pip install pytest
+poetry install
+poetry run pytest
+```
+
+## Python Automation Framework
+* [python-automation-framework](https://github.com/mreiche/python-automation-framework)
+
+Steps to run tests on a docker container:
+```
+git clone https://github.com/mreiche/python-automation-framework.git
+cd python-automation-framework
+docker run -it --name python-automation-framework -w /opt -v$(pwd):/opt python:3.11 bash
+pip install pytest
+PYTHONPATH="." pytest --numprocesses=4 --cov=paf test
```

Comment:
I get:

ImportError while loading conftest '/opt/test/conftest.py'.
test/__init__.py:4: in <module>
    import inject
E   ModuleNotFoundError: No module named 'inject'



Have you maybe forgotten to add

pip install -r requirements.txt

?

Reply:
Yes.. I'll add it